source/simulator/assemblers/newton_stokes.cc

```diff
@@ -0,0 +1,677 @@
+                {
+                  testing = false;
+                  std::cout << sample << " Not SPD: " << abc << "; " << std::endl;
+
+                  for (unsigned int i=0; i<stokes_dofs_per_cell; ++i)
+                  {
+                    for (unsigned int j=0; j<stokes_dofs_per_cell; ++j)
+                      std::cout << std::setprecision(1)  << data.local_matrix(i,j) << "," << std::flush;
+                    std::cout << "},{" << std::endl;
+                  }
+                  std::cout << std::endl;
+                  std::cout << std::setprecision(6) << std::endl;
+                  for (unsigned int i=0; i<stokes_dofs_per_cell; ++i)
+                  {
+                    for (unsigned int j=0; j<stokes_dofs_per_cell; ++j)
+                    {
+                      //std::cout << i << "," << j << "=" << data.local_matrix(i,j) << ", gpui = " << scratch.grads_phi_u[i] << ", eta = " << eta << ", gpuj = " << scratch.grads_phi_u[j] << ", deds = " << viscosity_derivative_wrt_strain_rate << ", ps = " << pressure_scaling << ", sr = " << strain_rate << ", " << std::sqrt(0.5*deviator(strain_rate)*deviator(strain_rate)) << std::endl;
+                      /*data.local_matrix(i,j) += ( // using tensor thing is complecating everything quite a bit, so removed it for now.
+                          scratch.grads_phi_u[i] * 2.0 * eta * scratch.grads_phi_u[j]
+                                                 + alpha * (scratch.grads_phi_u[i] * (viscosity_derivative_wrt_strain_rate * scratch.grads_phi_u[j]) * strain_rate
+                                                     + scratch.grads_phi_u[j] * (viscosity_derivative_wrt_strain_rate * scratch.grads_phi_u[i]) * strain_rate)
+                                                     - (pressure_scaling *
+                                                         scratch.div_phi_u[i] * scratch.phi_p[j])
+                                                         // finally the term -div(u). note the negative sign to make this
+                                                         // operator adjoint to the grad(p) term
+                                                         - (pressure_scaling *
+                                                             scratch.phi_p[i] * scratch.div_phi_u[j]))
+                                                             * scratch.finite_element_values.JxW(q);* /
+                      Assert(dealii::numbers::is_finite(data.local_matrix(i,j)),ExcMessage ("Error: Assembly matrix is not finite."));
+                    }
+                  }
+
+                  Assert(testing,ExcMessage ("Error: Assembly not SSPD!."));
+                }else
+                {
+                  //if(abc != 0)
+                  //  std::cout << abc << " " << std::flush;
+                }
+              }
+              if(testing == false)
+                std::cout << std::endl;
+              }*/
+
+              ////// end testing //////
```

Comment:
So @bangerth said we could make this (the spd checking part) into a debug only assert, but it might be a bit expensive for that (I am not sure how expensive it is any more, maybe it is not too bad).